File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        
        Arrays.sort(arr);
        List<List<Integer>>list=new ArrayList<>();
        int mini=Integer.MAX_VALUE;

        for(int i=1;i<arr.length;i++){
         mini=Math.min(mini,(arr[i]-arr[i-1]));
        }
     for(int i=0;i<arr.length-1;i++){
      
        for(int j=i+1;j<arr.length;j++){
           
            if((arr[j]-arr[i])==mini){
                  list.add(Arrays.asList(arr[i],arr[j]));
            }

        }
       

     }

return list;
    }

}

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        vector<vector<int>> ans;
        sort(arr.begin(),arr.end());
        unordered_set<int> st;
        int req=arr[1]-arr[0];
        st.insert(arr[0]);
        int n=arr.size();
        for(int i=1;i<n;i++)
        {
            st.insert(arr[i]);
            req=min(arr[i]-arr[i-1],req);
        }
        for(int i=0;i<n;i++)
        {
            int num=arr[i];
            int need=num+req;
            if(st.find(need)!=st.end())
            {
                ans.push_back({num,need});
            }
            
        }
        return ans;
        
    }
};